Marlin v1

green-striped-gecko opened the pull request:
Hi Daid,
this is just a test if I understood the idea. If I think I have contributed something of value I would send you a pull request and then you could review it, true? One thing I have not managed to understand, how do I "upgrade" my fork with improvements from your side? I put you on my watch list so I think I will be notified if you do a commit?

Sorry to bother you so much with such basic questions.

green-striped-gecko committed:
Progressbar and Autohome now with 5 mm in Z axis

green-striped-gecko committed:
Changed the rm message

to avoid user are using this fork without knowing the danger...

daid commented:
Yes, this would be the proper way to send in patches. Note that all changes you do on your fork automaticly get added after you created the pull request, so if you do another commit this commit will be added to the pull request. (you can avoid this by using branches, but I haven't used that yet)

The "fork a repo": https://help.github.com/articles/fork-a-repo
Explained that you had to setup an "remote" called "origin". This origin is used to get changes from my fork back into yours. With "git fetch upstream"

You pushed your new configuration.h to the fork, this happens quite a lot, but makes pull requests harder to merge, as we have to fix the configuration.h again after merging the pull request. So it's best to just commit the changes that you want to send.

green-striped-gecko commented:
Hi Daid
Thanks for explaining these things.
I guess easiest is to include the configuration.h into the gitignore file.
One last question could I fetch your latest changes also simply stating
git fetch https://...(your repository address) as i did create an upstream remote but this
Inks to my repo. There is also an origin name when i open gitgui so i assume that links to your repo. I try to find a way to check this.

Cheers,
Bernd

# 

Dr Bernd Gruber       Tel: (02) 6206 3804 Fax: (02) 6201 2328
Assistant Professor
Institute for Applied Ecology
Faculty of Applied Science
University of Canberra   ACT 2601 AUSTRALIA
Email: bernd.gruber@canberra.edu.aumailto:bernd.gruber@canberra.edu.au
WWW: http://www.canberra.edu.au/faculties/science/staff/profiles/dr-bernd-gruber

Australian Government Higher Education Provider Number CRICOS:#00212K

NOTICE & DISCLAIMER: This email and any files transmitted with it may contain
confidential or copyright material and are for the attention of the addressee
only. If you have received this email in error please notify us by email
reply and delete it from your system. The University of Canberra accepts
no liability for any damage caused by any virus transmitted by this email.

# 

On 07/09/2012, at 23:46, "daid" <notifications@github.com<mailto:notifications@github.com>> wrote:

Yes, this would be the proper way to send in patches. Note that all changes you do on your fork automaticly get added after you created the pull request, so if you do another commit this commit will be added to the pull request. (you can avoid this by using branches, but I haven't used that yet)

The "fork a repo": https://help.github.com/articles/fork-a-repo
Explained that you had to setup an "remote" called "origin". This origin is used to get changes from my fork back into yours. With "git fetch upstream"

You pushed your new configuration.h to the fork, this happens quite a lot, but makes pull requests harder to merge, as we have to fix the configuration.h again after merging the pull request. So it's best to just commit the changes that you want to send.

—
Reply to this email directly or view it on GitHubhttps://github.com/daid/Marlin/pull/5#issuecomment-8365614.

green-striped-gecko committed:
timer stops at the end of print,  (that is the info most user want to know)
visual feedback to submenues when clicked (you always know what will change)
changed partial release message to print finished, (did not make sense to me)
sd card shows reversed order of files (is quicker and most often the last file is the first to print)

green-striped-gecko commented:
Hi Daid, 

sorry I am still not sure how not to commit configuration.h . I included it into my gitignore file, not sure if that helps.
Cheers, Bernd

green-striped-gecko committed:
moved progress bar to message line

if people have a heated bed, the progress bar was conflicting with the temperature

green-striped-gecko commented on Marlin/ultralcd.pde:
```diff
@@ -2323,7 +2401,7 @@ void MainMenu::showSD()
         {
           BLOCK
-          card.getfilename(i-FIRSTITEM);
+          card.getfilename(nrfiles +1-i);
```
This reverses the order of files shown.

green-striped-gecko commented:
Hi Daid,

just a quick question. Is there a way to undo the earlier pull request and as you suggested to only submit the changes of files which are needed?

Cheers, Bernd

daid commented:
Not 100% sure, but you could revert the earlier commits. You could also make a separate branch for the pull request I guess. Because every commit on a branch (the mainline is also a branch) is put into the pull request.

But the easiest way (and what I would do myself, so I would be sure it works) is to delete your fork and start a new fork, in in that fork only commit the changes that you want to put in the pull request.

green-striped-gecko commented:
Hi Daid,

disregard this pull request. It was as you know the one which also included unnecessary files.
